### ISSUE TITLE
can't save a categorie with None as value

### ISSUE THREAD

**Issue by kain88-de** [datreant/datreant.core#117]
For me `None` is a normal python value like any other. For example I'm using datreant right now to store some processing arguments like start,step and stop. There `None` is a valid value.

**Comment by kain88-de** [comment]
This would also mean that setting a category to `None` would now change it's value.

**Comment by kain88-de** [comment]
I've started to change my position on this. I can actually do fit my case more if I add some data as `args` that have all the step/stop and everything set correctly. I can't really search with the categories anymore but I can live with that.

But I can still see some value in this. Or at least we should throw an error instead of silence error

**Comment by dotsdl** [comment]
This slipped under my radar, so apologies for the delay.

``None`` isn't allowed for a category value because ``None`` is used to indicate absence of a value for a category when accessed from a ``Bundle`` of ``Treant``s. Basically, if we have a ``Bundle`` of 5 ``Treant``s and all but the last one have a "huggable" category, then doing:

```python
b.categories['huggable']
```
will yield a list giving the values for each. For the last one, ``None`` will be given because it simply doesn't have the category key at all. If ``None`` is allowed as a valid value, this result is ambiguous.

If you can think of an alternative to how things work above, I'd be happy for ``None`` to be an allowed category value.

**Comment by orbeckst** [comment]
Although I get the sentiment that it appears weird that you cannot store `None`, I think the semantic meaning of the current implementation is ok and `category=None` really seems to be the only sensible way to explicitly indicate the absence of an associated category. If this is documented then I would not do anything else. Maybe add a warning box to the docs?

**Comment by dotsdl** [comment]
We've got one near the bottom of the [categories section](http://datreant.readthedocs.io/en/develop/tags-categories.html#using-categories).

**Comment by orbeckst** [comment]
Then I'd close this issue as won't fix – but @kain88-de might have a different opinion.

**Comment by kain88-de** [comment]
The current behavior though gives a silent error for `t.categories['foo'] = None`. There are two ways this line could behave. 

1. raise a TypeError saying that `None` is not a possible type
2. delete entry `'foo'` from the categories / raise warning when key doesn't exist *deleting non existing key*

**Comment by orbeckst** [comment]
I think the second behavior is more sane. 

--
Oliver Beckstein
email: orbeckst@gmail.com

**Comment by dotsdl** [comment]
@kain88-de I like (2), and actually I'd like to go further. Instead of getting a ``KeyError`` when trying access a category that doesn't exist in a ``Treant``, let's return ``None``. This would be consistent with how missing categories are represented in a ``Bundle``, and would actually allow you to still do the kind of thing you described at the top of the issue.

Basically, ``None`` would behave in a sense as a valid value, but would never actually be stored if that is the value. But retrieving it would always give the expected result.

Is this a wonky idea, or should we roll with it?

**Comment by kain88-de** [comment]
Yeah it makes sense to me. So we just decide that `None` is a missing value and my usecase as stated in the first comment is just misusing that.
